v0/destinations/hs/util.js

```diff
@@ -17,7 +131,183 @@ const getEmailAndUpdatedProps = properties => {
   };
 };
 
+/* NEW API util functions */
+
+/**
+ * look for the contact in hubspot and extract its contactId for updation
+ * Ref - https://developers.hubspot.com/docs/api/crm/contacts#endpoint?spec=GET-/crm/v3/objects/contacts
+ * @param {*} destination
+ * @returns
+ */
+const searchContacts = async (message, destination) => {
+  const { Config } = destination;
+  let searchContactsResponse;
+  let contactId;
+  const traits = getFieldValueFromMessage(message, "traits");
+  let propertyName;
+
+  // lookupField key provided in Config.lookupField not found in traits
+  // then default it to email
+  if (!traits[`${Config.lookupField}`]) {
+    propertyName = "email";
+  } else {
+    // look for propertyName (key name) in traits
+    // Config.lookupField -> lookupField
+    // traits: { lookupField: email }
+    propertyName = traits[`${Config.lookupField}`];
+  }
+
+  // extract its value from the known propertyName (key name)
+  // if not found in our structure then look for it in traits
+  // Config.lookupField -> lookupField
+  // eg: traits: { lookupField: email, email: "test@test.com" }
+  const value =
+    getFieldValueFromMessage(message, propertyName) ||
+    traits[`${propertyName}`];
+
+  if (!value) {
+    throw new CustomError(
+      `[HS] Identify:: '${propertyName}' lookup field not found in traits for contact lookup`,
+      400
+    );
+  }
+
+  const requestData = {
+    filterGroups: [
+      {
+        filters: [
+          {
+            propertyName,
+            value,
+            operator: "EQ"
+          }
+        ]
+      }
+    ],
+    sorts: ["ascending"],
+    properties: [propertyName],
+    limit: 2,
+    after: 0
+  };
+
+  if (Config.authorizationType === "newPrivateAppApi") {
+    // Private Apps
+    const requestOptions = {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${Config.accessToken}`
+      }
+    };
+    searchContactsResponse = await httpPOST(
+      IDENTIFY_CRM_SEARCH_CONTACT,
+      requestData,
+      requestOptions
+    );
+    searchContactsResponse = processAxiosResponse(searchContactsResponse);
+  } else {
+    // API Key
+    const url = `${IDENTIFY_CRM_SEARCH_CONTACT}?hapikey=${Config.apiKey}`;
+    searchContactsResponse = await httpPOST(url, requestData);
+    searchContactsResponse = processAxiosResponse(searchContactsResponse);
+  }
+
+  if (searchContactsResponse.status !== 200) {
+    // check if exists err.response && err.response.status else 500
+    if (searchContactsResponse.response) {
```

Comment:
You don't need to handle the presence of  `searchContactsResponse.response` it will be handled in `processAxiosResponse`